url60801aaae5_yurgenbeerman_JavaOOP_with_Maven_tgz-pJ8-edu_services_PublicServiceDemoJ8/src/edu/services/docs/OrganizationDocument.java

```diff
@@ -0,0 +1,227 @@
+package edu.services.docs;
+
+import edu.clients.DocumentCreator;
+import edu.services.orgs.PublicService;
+import edu.services.orgs.PublicServiceDepartment;
+import edu.utils.PublicRequestsUtils;
+
+import static edu.services.docs.DocDefaults.*;
+
+import java.security.InvalidParameterException;
+import java.util.GregorianCalendar;
+
+/**
+ * Created by yurii.pyvovarenko on 3/4/14.
+ */
+public class OrganizationDocument extends Text {
+    private static long lastDocumentId;
+
+    private DocumentLifecycle documentLifecycle;
+    private long documentId;
+    private String documentName;
+    private long documentAuthorId;
+    private DocumentCreator author;
+    private GregorianCalendar documentCreationDate;
+    private DocumentType documentType;
+    private String documentNumber;
+    private DocumentStatus documentStatus;
+    private long orgId;
+    private String text;
+    private PublicServiceDepartment currentPublicServiceDepartment;
+
+    public OrganizationDocument() {
+        this.documentId = OrganizationDocument.lastDocumentId;
+        OrganizationDocument.lastDocumentId++;
+    }
+
+    public OrganizationDocument(DocumentType documentType, DocumentCreator author, PublicService publicService) {
+        this();
+        if (documentType != null) {
+            this.documentType = documentType;
+        } else {
+            throw new InvalidParameterException("documentType");
+        }
+        if (author != null) {
+            this.author = author;
+        } else {
+            throw new InvalidParameterException("author");
+        }
+        if (publicService != null) {
+            this.orgId = publicService.getOrgId();
+        } else {
+            throw new InvalidParameterException("publicService");
+        }
+        this.documentNumber = documentType.getDocTypeShortName() + this.documentId;
+        this.documentStatus = new DocumentStatus(documentType.getDocumentLifecycle());
+        this.setDocumentCreationDate(documentStatus.getZeroStatusDate());
+        this.setDocumentAuthorId(author.getId());
+    }
+
+    public String getValidityString() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 73-81, lines 101-107 into separate methods.